src/App.js

```diff
@@ -1,23 +1,93 @@
+  const [remainingContacts, setRemainingContacts] = useState(contactsData.slice(5));
+
+
+  const addRandomContact = () => {
+    if (remainingContacts.length === 0) {
+      alert("No more contacts to ad`d!");
+      return;
+    }
+    const randomIndex = Math.floor(Math.random() * remainingContacts.length);
+    const randomContact = remainingContacts[randomIndex];
+    setContacts((prevState) => [...prevState, randomContact]);
+    setRemainingContacts((prevState) => prevState.filter(contact => contact.id !== randomContact.id))
+  };
+
+  const sortByName = () => {
+    const sortedContacts = [...contacts].sort((a, b) => {
+      return a.name.localeCompare(b.name);
+    });
+    setContacts(sortedContacts);
+  };
+
+  const sortByPopularity = () => {
+    const sortedContacts = [...contacts].sort((a, b) => {
+      return b.popularity - a.popularity;
+    });
+    setContacts(sortedContacts);
+  };
+
+  const deleteContact = (id) => {
+    const updatedContacts = contacts.filter((contact) => contact.id !== id);
+    setContacts(updatedContacts);
+  };
+
   return (
     <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <h1>IronContacts</h1>
+      <div className="buttons">
+        <button className="mr-4" onClick={addRandomContact}>Add Random Contact</button>
+        <button className="mr-4" onClick={sortByPopularity}>Sort by popularity</button>
+        <button onClick={sortByName}>Sort by name</button>
+      </div>
+
+      <div className="table-container">
+        <table>
+          <thead>
+            <tr>
+              <th>Picture</th>
+              <th>Name</th>
+              <th>Popularity</th>
+              <th>
+                Won <br /> Oscar
+              </th>
+              <th>
+                Won <br /> Emmy
+              </th>
+              <th>Action</th>
+            </tr>
+          </thead>
+          <tbody>
+            {contacts.map((contact) => (
+              <tr key={contact.id}>
+                <td>
+                  <img
+                    src={contact.pictureUrl}
+                    alt={contact.name}
+                    height="100"
+                  />
+                </td>
+                <td>{contact.name}</td>
+                <td>{contact.popularity.toFixed(2)}</td>
+                <td>{contact.wonOscar ? <span>🏆</span> : null}</td>
+                <td>{contact.wonEmmy ? <span> 🌟</span> : null}</td>
+                <td>
+                  <button onClick={() => deleteContact(contact.id)}>
+                    Delete
+                  </button>
+                </td>
+              </tr>
```

Comment:
This hole section code be a new component that only handles the information of the contact, in this case you could do something like  
```suggestion
          <ContanctInfo  contact={contact} />
```